websocket/server.go

```diff
@@ -102,23 +104,17 @@ func (s *Server) serveMessages(w http.ResponseWriter, r *auth.AuthenticatedReque
 	}
 
 	// call the incomerHandler that will create the Speaker
-	c, err = s.incomerHandler(conn, r)
+	c, err = s.incomerHandler(conn, r, func(c *wsIncomingClient) (Speaker, error) { return c, nil })
```

Comment:
why do we need a promoter if it just returns the input value (plus error = nil).

Reply:
There are 2 implementations of the promoter : one that indeed returns the input value, the other that returns a `StructSpeaker`